File: net/merge/Merger_cgm_pdf.cs
namespace aspose_snippets.net
{
    public static partial class Merger
    {
        public static void cgm_to_pdf()
        {
            const string pathSource1 = "../../TestData/test.cgm";
            const string pathSource2 = "../../TestData/Second/test.cgm";

            //cgm files can be parsed and loaded as Aspose Document
            var firstDoc = new Aspose.Pdf.Document(pathSource1,
                new Aspose.Pdf.CgmLoadOptions());
            var secondDoc = new Aspose.Pdf.Document(pathSource2,
                new Aspose.Pdf.CgmLoadOptions());

            //create empty pdf document
            var outputDoc = new Aspose.Pdf.Document
            {
                //set less memory usage with unload instead of fast performance
                EnableObjectUnload = true
            };

            foreach (var page in firstDoc.Pages)
            {
                //add page from one document to another directly
                outputDoc.Pages.Add(page);
            }

            foreach (var page in secondDoc.Pages)
            {
                //add page from one document to another directly
                outputDoc.Pages.Add(page);
            }

            //save result pdf to file
            outputDoc.Save("Merger_cgm_pdf.pdf", Aspose.Pdf.SaveFormat.Pdf);
        }
    }
}

File: java/merge/src/aspose_snippets/java/Merger_cgm_pdf.java
package aspose_snippets.java;

public class Merger_cgm_pdf {
    public static void Execute() {
        String pathSource1 = "../../TestData/test.cgm";
        String pathSource2 = "../../TestData/Second/test.cgm";

        //cgm files can be parsed and loaded as Aspose Document
        com.aspose.pdf.Document firstDoc = new com.aspose.pdf.Document(pathSource1,
                new com.aspose.pdf.CgmLoadOptions());
        com.aspose.pdf.Document secondDoc = new com.aspose.pdf.Document(pathSource2,
                new com.aspose.pdf.CgmLoadOptions());

        //create empty pdf document
        com.aspose.pdf.Document outputDoc = new com.aspose.pdf.Document();

        //set less memory usage with unload instead of fast performance
        outputDoc.setEnableObjectUnload(true);

        for (com.aspose.pdf.Page page : firstDoc.getPages()) {
            //add page from one document to another directly
            outputDoc.getPages().add(page);
        }

        for (com.aspose.pdf.Page page : secondDoc.getPages()) {
            //add page from one document to another directly
            outputDoc.getPages().add(page);
        }

        //save result pdf to file
        outputDoc.save("Merger_cgm_pdf.pdf", com.aspose.pdf.SaveFormat.Pdf);
    }
}
